Enclose pre-read with Read-all; de-dupe plugin table and show types, etc

fredjk-gh opened the pull request:
A number changes in preparation for the upcoming Announcer plugin.
1) De-dupe entries in the plugin table in the core settings UI when a plugin implements both Worker and Notifier interfaces.
2) Also add a "Types" column to the plugin table indicating the set of implemented interfaces found.
3) Wrap the pre-reading logic in "Read-all" scope to properly suppress  a stampede of notifications (particularly audio ones) when starting the log monitor.
4) Fix the int setting UI. It rendered incorrectly, it didn't get the value from the settings class, nor wrote it back. Also implemented optional value bounding using the new setting attribute (see the related Framework PR). 
5) Applied similar fixes to string setting (addressed rendering, write-back although untested).
6) Minor visual fixes (height/vertical alignment) for non-boolean settings to address cut-off/overlapping inputs.

Sorry for the larger PR, but these *should* be all the loose ends before we could consider including Basic Announcer...

The screencap below shows the current state of core settings UI.

![image](https://user-images.githubusercontent.com/54195004/129433764-9abd107b-453e-4bf1-a001-b928e3da416a.png)

Xjph reviewed (commented):
Holding off on accepting this one for now, since it uses features from a framework PR that's not approved yet.

Getting the `int` and `string` type settings working is obviously a good change, and I'm fine with the read-all tweaking as well. Though that probably should've been two separate PRs. 

I'm not completely sold on having a new `enum` that's used basically once ever, but it does highlight a need to have a better way to easily check whether a plugin is a worker or notifier, so this gets a little "approved, but possibly revisit later" flag in my head.

Will come back and approve formally once xjph/ObservatoryFramework#3 is resolved.

fredjk-gh commented:
> Holding off on accepting this one for now, since it uses features from a framework PR that's not approved yet.
Acknowledged. I've responded on that PR now. 

> Getting the `int` and `string` type settings working is obviously a good change, and I'm fine with the read-all tweaking as well. Though that probably should've been two separate PRs.
So, I've now implemented a Slider based numeric picker *in addition* to fixing the NumericUpDown implementation. As such, I've reverted out the readall tweaks to keep this focused and send a separate PR. I'll update this PR description momentarily.

> I'm not completely sold on having a new `enum` that's used basically once ever, but it does highlight a need to have a better way to easily check whether a plugin is a worker or notifier, so this gets a little "approved, but possibly revisit later" flag in my head.
Fair enough. For the moment I've backed out the single use enum and simply used the interface names (eg. `typeof(IObservatoryWorker).Name`) for the respective types for the time being. We can revisit later.

> Will come back and approve formally once [Xjph/ObservatoryFramework#3](https://github.com/Xjph/ObservatoryFramework/pull/3) is resolved.
